src/fondant/component/executor.py

```diff
@@ -298,7 +300,7 @@ def _run_execution(
         )
         self._write_data(dataframe=output_df, manifest=output_manifest)
 
-        component.teardown()
+        component.teardown(state)
```

Comment:
Shouldn't this be `component.teardown(*state)` in case someone returns more as a single state?

Reply:
This wouldn't work for a single object. I think it's most clear if we just inject exactly what has been returned by the setup method. If it returns a tuple, we inject a tuple.